Desktop_Interface/functiongencontrol.cpp

```diff
@@ -142,36 +72,37 @@ void functionGenControl::waveformName_CH2(QString newName)
 
     line = fptr.readLine();
     strcpy(lengthString, line.data());
-    sscanf(lengthString, "%d", &length_CH2);
+    sscanf(lengthString, "%d", &channel.length);
     qDebug() << "lengthString" << lengthString;
 
     line = fptr.readLine();
     strcpy(divisibilityString, line.data());
-    sscanf(divisibilityString, "%d", &divisibility_CH2);
+    sscanf(divisibilityString, "%d", &channel.divisibility);
     qDebug() << "divisibilityString" << divisibilityString;
 
-    qDebug() << "Length = " << length_CH2;
-    qDebug() << "Divisibility = " << divisibility_CH2;
+    qDebug() << "Length = " << channel.length;
+    qDebug() << "Divisibility = " << channel.divisibility;
 
     QByteArray remainingData = fptr.readAll();
     char *dataString = remainingData.data();
 
-    free(samples_CH2);
-    samples_CH2 = (unsigned char *) malloc(length_CH2);
+    free(channel.samples);
+    channel.samples = (unsigned char *) malloc(channel.length);
 
     int dummy;
     char *dataStringCurrent = dataString;
-    for (int i=0;i<length_CH2;i++){
+    for (int i=0;i<channel.length;i++){
         sscanf(dataStringCurrent, "%d", &dummy);
         dataStringCurrent += strcspn(dataStringCurrent, "\t") + 1;
-        samples_CH2[i] = (unsigned char) dummy;
+        channel.samples[i] = (unsigned char) dummy;
     }
+
 #else
     QString directory = QCoreApplication::applicationDirPath();
 
     directory.append("/waveforms/");
     directory.append(newName);
-    QByteArray temp = directory.toLatin1();
+    QByteArray temp = encode(directory);
```

Comment:
This is another parameter than can be removed from the template (soon to be method) - either UTF-8 or the native encoding (which I suspect for most locales would be UTF-8) should always be used as it solves a whole lot of issues related to users having Chinese/Japanese/Korean etc. characters in file names and paths.  At work, it's one of QA's favourite things to do - can we break the software with a Japanese file name?

Reply:
Ahhh that's a good point. I was refactoring without putting much thought into the reason of things so i just parameterized the different functionality and didn't even consider that, but now that you mention it, it feels obvious.